renovate/create-wrapper.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -euo pipefail
+
+# Script to create a Renovate wrapper that auto-exports RENOVATE_* secrets
+# Usage: create-wrapper.sh <output-path>
+# Note: Expects SECRETS_JSON environment variable to be set
+
+OUTPUT_PATH="$1"
+
+echo "🔧 Creating Renovate wrapper script..."
+
+# Validate that SECRETS_JSON is set
+if [ -z "${SECRETS_JSON:-}" ]; then
+  echo "❌ Error: SECRETS_JSON environment variable is not set"
+  exit 1
+fi
+
+# Clean and validate the JSON using jq (this removes control characters)
+echo "🔍 Cleaning and validating JSON..."
+CLEANED_JSON=$(echo "$SECRETS_JSON" | jq -c . 2>&1) || {
+  echo "❌ Error: Invalid JSON in SECRETS_JSON"
+  exit 1
+}
+
+echo "✅ JSON cleaned and validated"
+
+# Encode to base64 (use -w 0 to disable line wrapping)
+SECRETS_BASE64=$(echo "$CLEANED_JSON" | base64 -w 0)
+
+echo "✅ Secrets validated and encoded"
+
+# Create the wrapper script with base64-encoded secrets
+cat > "$OUTPUT_PATH" <<WRAPPER_EOF
+#!/bin/bash
+set -euo pipefail
+
+# Decode secrets from base64
+ALL_SECRETS=\$(echo "${SECRETS_BASE64}" | base64 -d)
+
+echo "🔧 Auto-detecting and exporting RENOVATE_* secrets..."
+echo ""
```

Comment:
Is it worthwhile to validate we're not overriding any `RENOVATE_` env variable already present in the environment?